### PR TITLE
Added MixedKernelSeparateMof

### PR DESCRIPTION
**Pull request by Joshuaalbert**
## PR content:

* [Title](#title)
* [Description](#description)
* [Minimal working example](#minimal-working-example)

### Title

This provides the missing MixedKernelSeparateMof that does efficient  mixing of ind. latents for separate inducing points.

### Description

Working on the solution to #819 I found that one way to address the problem was to define the missing feature MixedKernelSeparateMof and to create the corresponding dispatch route. Bug #819 is still not resolved because the path `conditional(feat: SeparateIndependentMof, kern: SeparateMixedMok)` has a shape conflict, however this PR will add a more efficient route anyways.

### Minimal working example

See the commited `notebooks/multioutput.ipynb`

## Code quality requirements

Not a significant code change

### CODE REVIEW

**Comment by Joshuaalbert** [comment]
@st-- and @markvdw can this be merged?

File: gpflow/multioutput/features.py [hunk 1]
```diff
@@ -167,7 +178,7 @@ def Kuu(feat, kern, *, jitter):
     return Kmm + jittermat
 
 
-@dispatch(SeparateIndependentMof, (SeparateIndependentMok, SeparateMixedMok))
+@dispatch((SeparateIndependentMof,MixedKernelSeparateMof), (SeparateIndependentMok, SeparateMixedMok))
```
Comment (st--) [inline gpflow/multioutput/features.py]: Is this supposed to be an implementation for the outer product of (SeparateIndependentMof,MixedKernelSeparateMof) and (SeparateIndependentMok, SeparateMixedMok)? I.e. is this correct for Kuu(SeparateIndependentMof, SeparateMixedMok) and Kuu(MixedKernelSeparateMof, SeparateIndependentMok)?

**Comment by st--** [comment]
@Joshuaalbert looks good to me, would be good if you could clarify the one comment I left.

Can you think of a good way of testing the Kuu method you added ?

File: gpflow/multioutput/features.py (hunk 1 above)
Comment (Joshuaalbert) [reply gpflow/multioutput/features.py]: Yes. Prior to my addition it was already `SeparateIndependentMof, (SeparateIndependentMok, SeparateMixedMok)` which simply zips along both the feature list and kernel list and stack into shapes `LxMxN` for `Kuf` and `LxMxM` for `Kuu`. The same behaviour is required for any scenario where there are multiple features and multiple kernels, which is true of any combination of  `((SeparateIndependentMof,MixedKernelSeparateMof), (SeparateIndependentMok, SeparateMixedMok))`.

**Comment by vdutor** [comment]
LGTM. Could you please add a test for your new class `MixedKernelSeparateMof` to https://github.com/GPflow/GPflow/blob/develop/tests/test_multioutput.py#L452?

**Comment by Joshuaalbert** [comment]
I've added a test now. 
``` python
def test_MixedKernelSeparateMof(session_tf):
    data = DataMixedKernel

    kern_list = [RBF(data.D) for _ in range(data.L)]
    feat_list = [InducingPoints(data.X[:data.M, ...].copy()) for _ in range(data.L)]
    k1 = mk.SeparateMixedMok(kern_list, W=data.W)
    f1 = mf.SeparateIndependentMof(feat_list)
    m1 = SVGP(data.X, data.Y, k1, Gaussian(), feat=f1, q_mu=data.mu_data, q_sqrt=data.sqrt_data)

    kern_list = [RBF(data.D) for _ in range(data.L)]
    feat_list = [InducingPoints(data.X[:data.M, ...].copy()) for _ in range(data.L)]
    k2 = mk.SeparateMixedMok(kern_list, W=data.W)
    f2 = mf.MixedKernelSeparateMof(feat_list)
    m2 = SVGP(data.X, data.Y, k2, Gaussian(), feat=f2, q_mu=data.mu_data, q_sqrt=data.sqrt_data)

    check_equality_predictions(session_tf, [m1, m2])
```
However, this will fail until #819 is resolved because the proper comparison test is (SeparateIndependentMof, SeparateMixedMok) v.s. (MixedKernelSeparateMof,SeparateMixedMok) and the first path is broken.

**Comment by st--** [comment]
@Joshuaalbert Hi would you be willing to port this across to gpflow2 ?

**Pull request merged**